### PR TITLE
Update postgresql connector to 42.0.0

### PR DESCRIPTION
**Pull request by nerzhul**
postgresql is very very very old (PostgreSQL 9.1). It's not compatible with postgresql 9.6, update to a postgresql 9.6 version with JDK 8 support.

See https://jdbc.postgresql.org/download.html

### CODE REVIEW

File: rundeckapp/grails-app/conf/BuildConfig.groovy [hunk 1]
```diff
@@ -123,7 +123,7 @@ grails.project.dependency.resolution = {
         compile("org.rundeck:rundeck-core:${rundeckVersion}")
         compile("org.rundeck:rundeck-storage-filesys:${rundeckVersion}")
 
-        runtime 'postgresql:postgresql:9.1-901.jdbc4'
+        runtime 'postgresql:postgresql:42.0.0'
```
Comment (gschueler) [inline rundeckapp/grails-app/conf/BuildConfig.groovy]: note: i think the dependency group id changed to `org.postgresql` so it should be:

    runtime 'org.postgresql:postgresql:42.0.0'

**Comment by nerzhul** [comment]
Fixed

**Comment by gschueler** [comment]
thanks!

**Comment by nerzhul** [comment]
no problem

**Pull request merged**